### DISCUSSION TITLE
Lanczos interpolation proposal

### DISCUSSION THREAD

**Discussion by gazzar** [SarthakJariwala/seaborn-image#254] [Ideas]
I guess the philosophy of this promising package is to choose the best image display defaults. What do you think about the idea of defaulting to using lanczos interpolation instead of None, which is backend dependent (typically nearest-neighbour). For teaching image processing, I have found that writing an imshow wrapper that defaults to using lanczos interpolation generates the "least surprising" image, particularly for large scaled-down images.

**Comment by SarthakJariwala** [comment]
Thanks for the proposal, @gazzar ! I am definitely open to it.
One question since I do not have experience using the lanczos interpolation method - Is there any particular reason why the new default should be lanczos and not something like hanning, etc?

**Reply by gazzar** [reply]
Good question. Unfortunately I'm not an expert in this either. I can say a couple of things about why I chose Lanczos over the other interpolation schemes for my own imshow wrapper; I evaluated all the interpolation schemes matplotlib provides and decided that it gave the truest representation of the features in the image data within Jupyter notebooks, which were scaled down versions of the original. Wikipedia cites a reference advocating for this choice https://en.wikipedia.org/wiki/Lanczos_resampling '... to resize or rotate a digital image. It has been considered the "best compromise" among several simple filters for this purpose.' Beyond those two reasons, I don't know the circumstances under which one might choose one of the many other filters provided. It is possible something else might also be a good choice.
